apps/dashboard/src/components/primitives/field-editor/field-editor.tsx

```diff
@@ -0,0 +1,171 @@
+import { EditorView } from '@uiw/react-codemirror';
+import { useMemo, useState, useRef, useCallback } from 'react';
+import { Completion, CompletionContext } from '@codemirror/autocomplete';
+
+import { Editor } from '@/components/primitives/editor';
+import { completions } from '@/utils/liquid-autocomplete';
+import { LiquidVariable } from '@/utils/parseStepVariablesToLiquidVariables';
+import { autocompletion } from '@codemirror/autocomplete';
+import { Popover, PopoverTrigger } from '@/components/primitives/popover';
+import { createVariablePlugin } from './variable-plugin';
+import { VariablePopover } from './variable-popover';
+import { variablePillTheme } from './variable-plugin/variable-theme';
+
+type FieldEditorProps = {
+  value: string;
+  onChange: (value: string) => void;
+  variables: LiquidVariable[];
+  placeholder?: string;
+  autoFocus?: boolean;
+  size?: 'default' | 'lg';
+  fontFamily?: 'inherit';
+  id?: string;
+};
+
+export const FieldEditor = ({
```

Comment:
A new component that wraps the <Editor component with the needed behaviour for content management